src/main/webapp/app/pages/usageAnalysisPage/UserUsageDetailsTable.tsx

```diff
@@ -0,0 +1,134 @@
+import OncoKBTable from 'app/components/oncokbTable/OncoKBTable';
+import { filterByKeyword } from 'app/shared/utils/Utils';
+import autobind from 'autobind-decorator';
+import _ from 'lodash';
+import { observable } from 'mobx';
+import { observer } from 'mobx-react';
+import React from 'react';
+import { Row, ToggleButton, ToggleButtonGroup } from 'react-bootstrap';
+import { UsageRecord } from 'app/pages/usageAnalysisPage/UsageAnalysisPage';
+import {
+  USAGE_DETAIL_TIME_KEY,
+  USGAE_ALL_TIME_KEY,
+} from 'app/config/constants';
+
+type IUserUsageDetailsTable = {
+  data: Map<string, UsageRecord[]>;
+  loadedData: boolean;
+  defaultResourcesType: number;
+  defaultTimeType: number;
+};
+
+@observer
+export default class UserUsageDetailsTable extends React.Component<
+  IUserUsageDetailsTable,
+  {}
+> {
+  @observable resourcesTypeToggleValue = this.props.defaultResourcesType;
+  @observable timeTypeToggleValue = this.props.defaultTimeType;
```

Comment:
We should type this variable. I would not understand what 1 represents and what 2.